src/parse.rs

```diff
@@ -1922,13 +1928,20 @@ pub(crate) struct HtmlScanGuard {
     pub declaration: usize,
 }
 
+pub struct BrokenLinkFix<'a> {
+    title: CowStr<'a>,
+    url: CowStr<'a>,
+}
+
+pub type BrokenLinkCallback<'a> = Option<&'a dyn Fn(BrokenLink) -> Option<BrokenLinkFix<'a>>>;
```

Comment:
While you're at it, can you change this to `FnMut` so it doesn't have to use `Cell` and `RefCell` for modifying state? I ran into trouble with this while writing the test case.
```suggestion
pub type BrokenLinkCallback<'a> = Option<&'a mut dyn FnMut(BrokenLink) -> Option<BrokenLinkFix<'a>>>;
```

Reply:
Yes, can do!